docs/cub/developer_overview.rst

```diff
@@ -121,26 +120,25 @@ An example of :cpp:struct:`cub::BlockReduce` demonstrates these patterns in prac
 Thread-level
 ************************************
 
-In contrast to algorithms at the warp/block/device layer,
-single threaded functionality like ``cub::ThreadReduce``
-is typically implemented as a sequential function and rarely exposed to the user.
+CUB thread-level algorithms are specialized for execution by a single CUDA thread.
 
 .. code-block:: c++
 
-    template <
-        int         LENGTH,
-        typename    T,
-        typename    ReductionOp,
-        typename    PrefixT,
-        typename    AccumT = detail::accumulator_t<ReductionOp, PrefixT, T>>
-    __device__ __forceinline__ AccumT ThreadReduce(
-        T           (&input)[LENGTH],
-        ReductionOp reduction_op,
-        PrefixT     prefix)
-    {
+    template <typename Input,
+              typename ReductionOp,
+              typename ValueT = ::cuda::std::__remove_cvref_t<decltype(::cuda::std::declval<Input>()[0])>,
+              typename AccumT = ::cuda::std::__accumulator_t<ReductionOp, ValueT>>
+    _CCCL_NODISCARD _CCCL_DEVICE _CCCL_FORCEINLINE
```

Comment:
Those should be
```suggestion
    [[nodiscard]] __device__
```

In documentation we always try to use the proper symbols and not our internal macros.

Also things like `__forceinline` are implementation details that only clutter the example